game/src/core/components/DrawComponent.java

```diff
@@ -2,143 +2,168 @@
+    private final Map<String, Animation> animationMap;
+    private final Logger LOGGER = Logger.getLogger(this.getClass().getName());
+    private Animation currentAnimation;
 
     /**
-     * Create a new AnimationComponent object
+     * Create a new DrawComponent.
+     *
+     * <p>Will read in all subdirectories of the given path and use each file in the subdirectory to
+     * create an animation. So each subdirectory should contain only the files for one animation.
      *
-     * <p>Create a new AnimationComponent with separate idle Animations for each direction.
+     * <p>Will set the current animation to idle left
      *
-     * @param entity Entity to add this component to
-     * @param idleLeft Idle-Animation faced left
-     * @param idleRight Idle-Animation faced right
+     * @param entity associated entity
+     * @param path Path (as a string) to the directory in the assets folder where the subdirectories
+     *     containing the animation files are stored. Example: "character/knight".
+     * @throws IOException if the given path does not exist
+     * @see Animation
      */
-    public DrawComponent(Entity entity, Animation idleLeft, Animation idleRight) {
+    public DrawComponent(Entity entity, String path) throws IOException {
         super(entity);
-        this.idleRight = idleRight;
-        this.idleLeft = idleLeft;
-        this.currentAnimation = idleLeft;
+        // fetch available animations
+        ClassLoader classLoader = getClass().getClassLoader();
+        File directory = new File(classLoader.getResource(path).getFile());
+        if (!directory.exists() || !directory.isDirectory()) {
+            throw new FileNotFoundException("Path " + path + " not found.");
+        }
+        animationMap =
+                Arrays.stream(directory.listFiles())
+                        .filter(File::isDirectory)
+                        .collect(Collectors.toMap(File::getName, Animation::of));
+        setCurrentAnimation(CoreAnimations.IDLE_LEFT);
     }
 
     /**
-     * Create a new AnimationComponent object.
+     * Create a new DrawComponent with a specific animation.
      *
-     * <p>Create a new AnimationComponent object with the given animation as idle animation for both
-     * directions.
+     * <p>The given animation will be used as idle-left and idle-right animation
      *
-     * @param entity Entity to add this component to
-     * @param idle Idle-Animation
+     * <p>This constructor is for special case only. Use {@link DrawComponent(Entity, String)} if
+     * possible.
+     *
+     * @param entity associated entity
+     * @param idle Animation to use as idle-left and idle-right animation.
      */
     public DrawComponent(Entity entity, Animation idle) {
-        this(entity, idle, idle);
+        super(entity);
+        animationMap = new HashMap<>();
+        animationMap.put(CoreAnimations.IDLE_LEFT.getPathString(), idle);
```

Comment:
Wenn wir die `toString` aus dem Enum nehmen würden, würden wir hier einen extra Methodenaufruf sparen, d.h. das wäre einfach `animationMap.put(CoreAnimations.IDLE_LEFT, idle);`. Die Lesbarkeit ist trotzdem gegeben aus meiner Sicht.

Das würde aber bedeuten, dass im Interface `IPath` tatsächlich `toString` gefordert wird (wie beim letzten Review). Fand ich eigentlich gut, finde ich immer noch besser.

Reply:
Hätte einen Nachgeschmack: `IPath` wäre kein funktionales interface mehr.
Scheint mit den Namen toString zu tun zu habe. Richtig checken tue ich das auch nicht.

Reply:
Aber brauchen wir an der Stelle wirklich ein funktionales I-face?

Reply:
Ich hab es jetzt nochmal getestet und auch mit `toString` brauchen wir diese aufrufe. 
So richtig sicher waurm, bin ich mir nicht. Ich vermute es hat irgendwas mit der "verschleierung" des Enums durch das Interface. zu tun.... 


Einfach der Zeit geschuldet: ich behalte den namen `getPathString ` bei, damit kann ich das implementieren der Methode noch enforcen.

Reply:
Ist gut. Wenn man die Methode sowieso aufrufen muss, dann kann sie auch besser so heissen. 

Und das kann man ja später auch immer nochmal in einem kleineren PR refactoren, wenn man möchte (und eine bessere Lösung hat).